smart-contract/hyperledger-fabric/v2/java/src/main/java/hu/bme/mit/ftsrg/tpcc/TPCC.java

```diff
@@ -86,10 +86,16 @@ public String doDelivery(Context ctx, String parameters) {
         // This is the oldest undelivered order of that district.
         // NO_O_ID, the order number, is retrieved.
 
-        NewOrder newOrder = LedgerUtils.getOldestNewOrder(ctx, params.w_id, d_id);
-        LOGGER.info("Oldest new order retrieved is: " + gson.toJson(newOrder));
+        NewOrder newO = new NewOrder();
+        newO.no_w_id = params.w_id;
+        newO.no_d_id = d_id;
+        List<NewOrder> orders = ctx.registry.readAll(ctx, newO);
+        NewOrder oldestNewOrder = orders.get(0);
```

Comment:
This is good, but as I said I don't think we can be certain that `Registry#readAll` will return the `NewOrder`s in the order of insertion. At this point, I would sort on the primary keys to be more robust.

Reply:
Also, now we are not handling the case when `readAll` returns an empty list (and `List#get` will throw an exception)